src/tokenizedSubaccounts/PPTSA.sol

```diff
@@ -0,0 +1,559 @@
+    assembly {
+      $.slot := PPTSAStorageLocation
+    }
+  }
+
+  constructor() {
+    _disableInitializers();
+  }
+
+  function initialize(
+    address initialOwner,
+    BaseTSA.BaseTSAInitParams memory initParams,
+    PPTSAInitParams memory ppInitParams
+  ) external reinitializer(2) {
+    __BaseTSA_init(initialOwner, initParams);
+
+    PPTSAStorage storage $ = _getPPTSAStorage();
+
+    $.baseFeed = ppInitParams.baseFeed;
+
+    $.depositModule = ppInitParams.depositModule;
+    $.withdrawalModule = ppInitParams.withdrawalModule;
+    $.rfqModule = ppInitParams.rfqModule;
+    $.optionAsset = ppInitParams.optionAsset;
+
+    BaseTSAAddresses memory tsaAddresses = getBaseTSAAddresses();
+
+    tsaAddresses.depositAsset.approve(address($.depositModule), type(uint).max);
+  }
+
+  ///////////
+  // Admin //
+  ///////////
+  function setPPTSAParams(PPTSAParams memory newParams) external onlyOwner {
+    // TODO: Add new params to this check. (What are some good bounds for these normally?)
+    if (
+      newParams.minSignatureExpiry < 1 minutes || newParams.minSignatureExpiry > newParams.maxSignatureExpiry
+        || (newParams.worstSpotBuyPrice < 1e18 || newParams.worstSpotBuyPrice > 1.2e18)
+        || (newParams.worstSpotSellPrice > 1e18 || newParams.worstSpotSellPrice < 0.8e18)
+        || (newParams.spotTransactionLeniency < 1e18 || newParams.spotTransactionLeniency > 1.2e18)
+        || newParams.optionVolSlippageFactor > 1e18
+        || newParams.minMarkValueToStrikeDiffRatio < newParams.maxMarkValueToStrikeDiffRatio
+        || newParams.maxMarkValueToStrikeDiffRatio > 1e20 || newParams.maxMarkValueToStrikeDiffRatio < 1e18
+        || newParams.optionMaxTimeToExpiry <= newParams.optionMinTimeToExpiry || newParams.optionMaxNegCash > 0
+        || newParams.feeFactor > 0.05e18
+    ) {
+      revert PPT_InvalidParams();
+    }
+
+    _getPPTSAStorage().ppParams = newParams;
+
+    emit PPTSAParamsSet(newParams);
+  }
+
+  ///////////////////////
+  // Action Validation //
+  ///////////////////////
+  function _verifyAction(IMatching.Action memory action, bytes32 actionHash) internal virtual override {
+    PPTSAStorage storage $ = _getPPTSAStorage();
+
+    if (
+      action.expiry < block.timestamp + $.ppParams.minSignatureExpiry
+        || action.expiry > block.timestamp + $.ppParams.maxSignatureExpiry
+    ) {
+      revert PPT_InvalidActionExpiry();
+    }
+
+    _revokeSignature($.lastSeenHash);
+    $.lastSeenHash = actionHash;
+
+    BaseTSAAddresses memory tsaAddresses = getBaseTSAAddresses();
+
+    if (address(action.module) == address($.depositModule)) {
+      _verifyDepositAction(action, tsaAddresses);
+    } else if (address(action.module) == address($.withdrawalModule)) {
+      _verifyWithdrawAction(action, tsaAddresses);
+    } else if (address(action.module) == address($.rfqModule)) {
+      _verifyTradeAction(action, tsaAddresses);
```

Comment:
we still want to keep the old tradeModule flow for collateral. Those shouldn't be going through RFQ

Reply:
Oh so you're suggesting an additional check for the trademodule? I'm not sure what that logic would be right now. I'll add the check in but will leave the implementation the same as CC. We can change this soon.